addons/bus/static/src/im_status_service.js

```diff
@@ -17,35 +20,51 @@ export const imStatusService = {
 
     start(env, { bus_service, multi_tab, presence }) {
         const imStatusModelToIds = {};
-        let updateBusPresenceTimeout;
-        const throttledUpdateBusPresence = timings.throttle(function updateBusPresence() {
-            clearTimeout(updateBusPresenceTimeout);
-            if (!multi_tab.isOnMainTab()) {
-                return;
-            }
+        const throttledUpdateBusPresence = timings.throttle(() => {
             const now = new Date().getTime();
             bus_service.send("update_presence", {
                 inactivity_period: now - presence.getLastPresence(),
                 im_status_ids_by_model: { ...imStatusModelToIds },
             });
-            updateBusPresenceTimeout = browser.setTimeout(
-                throttledUpdateBusPresence,
-                UPDATE_BUS_PRESENCE_DELAY
+        }, FETCH_IM_STATUS_THROTTLE);
+        let updateBusPresenceTimeout;
+        function updateBusPresencePeriodically() {
+            clearTimeout(updateBusPresenceTimeout);
+            if (!multi_tab.isOnMainTab()) {
+                return;
+            }
+            throttledUpdateBusPresence();
+            browser.setTimeout(
+                updateBusPresencePeriodically,
+                UPDATE_BUS_PRESENCE_DELAY + UPDATE_BUS_PRESENCE_JITTER * Math.random()
             );
-        }, UPDATE_BUS_PRESENCE_DELAY);
+        }
 
         bus_service.addEventListener("connect", () => {
             // wait for im_status model/ids to be registered before starting.
-            browser.setTimeout(throttledUpdateBusPresence, 250);
+            browser.setTimeout(updateBusPresencePeriodically, 500);
         });
-        multi_tab.bus.addEventListener("become_main_tab", throttledUpdateBusPresence);
-        bus_service.addEventListener("reconnect", throttledUpdateBusPresence);
+        multi_tab.bus.addEventListener("become_main_tab", updateBusPresencePeriodically);
+        bus_service.addEventListener("reconnect", updateBusPresencePeriodically);
         multi_tab.bus.addEventListener("no_longer_main_tab", () =>
             clearTimeout(updateBusPresenceTimeout)
         );
         bus_service.addEventListener("disconnect", () => clearTimeout(updateBusPresenceTimeout));
-
+        bus_service.subscribe("bus.bus/websocket_closed", () => {
+            throttledUpdateBusPresence.cancel();
+            updateBusPresencePeriodically();
+        });
         return {
+            /**
+             * Fetch the im_status that were previously registered.
+             *
+             * @param {Object} options
+             * @param {boolean} [options.waitForRegistration] Whether the fetch should be
+             * slighly delayed until the im_status model/ids are registered.
+             */
+            fetch({ waitForRegistration = false } = {}) {
+                setTimeout(throttledUpdateBusPresence, waitForRegistration ? 500 : 0);
```

Comment:
Ideally we should also reset the delay of `updateBusPresencePeriodically` when calling this one.